proto/penumbra/penumbra/core/asset/v1/asset.proto

```diff
@@ -92,6 +92,13 @@ message Value {
   AssetId asset_id = 2;
 }
 
+message Balance {
+  // Indicates if the balance is negated.
+  bool negated = 1;
+  // Represents the vector of 'Values' in the balance.
+  repeated Value balance = 2;
+}
```

Comment:
xref https://github.com/penumbra-zone/penumbra/pull/4943/files#r1855748563

Since `Value`s can't be negative, I think we'll want to have a list of pairs, each with a negation flag, something like
```
message Balance {
  message SignedValue {
    Value value = 1;
    bool negated = 2;
  }
  repeated SignedValue values = 1;
}
```
The top-level bool in the Rust `Balance` struct is an optimization that makes sign changes less expensive (unclear this was really necessary perf-wise, but, we have it now), IMO it should be excluded from serialization by having the serialized form always be "normalized" with individual sign bools.

Reply:
makes sense, refactored to normalize the top-level `Balance` and capture the sign information in the `SignedValue` pairs.

Reply:
how should we handle cases where multiple entries share the same asset ID with [different](https://github.com/penumbra-zone/penumbra/pull/4943/files#diff-9064ed98f9c25749e6785cab41b153fbd2e3cf99ab342f16abdc83d16d1f0f5aR63-R68) negation flags, for instance:

```ts
        let proto_balance = pb::Balance {
            values: vec![
                pb::balance::SignedValue { 
                    value: Some(pb::Value { asset_id: Some((*STAKING_TOKEN_ASSET_ID).into()), amount: Some(Amount::from(100u128).into()) }), 
                    negated: false
                },
                pb::balance::SignedValue {
                    value: Some(pb::Value {
                        asset_id: Some((*STAKING_TOKEN_ASSET_ID).into()),
                        amount: Some(Amount::from(200u128).into()),
                    }),
                    negated: true,
                },
            ],
        };

        let balance = Balance::try_from(proto_balance).expect("fallible conversion");
```

The `BTreeMap` can't hold multiple imbalances for the same asset ID. Instead, should it be doing something more sophisticated and explicitly accumulating values during conversion?